iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -238,52 +240,62 @@ def _adjust_operator_bundle(manifests_path, metadata_path, organization=None, pi
-            bundle_metadata = _get_bundle_metadata(operator_manifest, pinned_by_iib)
+            bundle_metadata = _get_bundle_metadata(
+                operator_manifest, pinned_by_iib, ignore_sanity_checks=True
+            )
             _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_replacements)
         elif customization_type == 'csv_annotations' and organization:
             org_csv_annotations = customization.get('annotations')
             if org_csv_annotations:
                 log.info('Applying csv annotations for organization %s', organization)
                 _adjust_csv_annotations(operator_manifest.files, package_name, org_csv_annotations)
+        elif customization_type == 'image_name_from_labels':
+            org_image_name_template = customization.get('template', '')
+            if org_image_name_template:
+                new_bundle_metadata = _get_bundle_metadata(operator_manifest, pinned_by_iib)
+                _replace_image_name_from_labels(new_bundle_metadata, org_image_name_template)
 
     return labels
 
 
-def _get_bundle_metadata(operator_manifest, pinned_by_iib):
+def _get_bundle_metadata(operator_manifest, pinned_by_iib, ignore_sanity_checks=False):
     """
     Get bundle metadata i.e. CSV's and all relatedImages pull specifications.
 
     :param operator_manifest.operator.OperatorManifest operator_manifest: the operator manifest
         object.
     :param bool pinned_by_iib: whether or not the bundle image has already been processed by
         IIB to perform image pinning of related images.
+    :param bool ignore_sanity_check: boolean specifying if sanity checks should be skipped.
+        defaults to ``False``
     :raises IIBError: if the operator manifest has invalid entries
     :return: a dictionary of CSV's and relatedImages pull specifications
     :rtype: dict
     """
     bundle_metadata = {'found_pullspecs': set(), 'operator_csvs': []}
     for operator_csv in operator_manifest.files:
-        if pinned_by_iib:
-            # If the bundle image has already been previously pinned by IIB, the relatedImages
-            # section will be populated and there may be related image environment variables.
-            # However, we still want to process the image to apply any of the other possible
-            # changes.
-            log.info('Skipping pinning because related images have already been pinned by IIB')
-        elif operator_csv.has_related_images():
-            csv_file_name = os.path.basename(operator_csv.path)
-            if operator_csv.has_related_image_envs():
-                raise IIBError(
-                    f'The ClusterServiceVersion file {csv_file_name} has entries in '
-                    'spec.relatedImages and one or more containers have RELATED_IMAGE_* '
-                    'environment variables set. This is not allowed for bundles regenerated with '
-                    'IIB.'
+        if ignore_sanity_checks:
```

Comment:
This seems to be the inverse of ignore. We should either rename this flag or invert the check.

Reply:
If you invert the check, also invert the calls to `_get_bundle_metadata`, of course.

Reply:
doh, my bad. Yes, I'll change it